Testscripts/Linux/validate-da.sh

```diff
@@ -220,11 +224,15 @@ function enable_disable_da(){
         fail_test "/dev/msda does not exist."
     fi
     # Check for "driver setup status=0" and "starting the dependency agent" in service.log.1 and service.log respectively
-    if ! grep -iq "driver setup status=0" $da_log_dir/service.log.1; then
+    if ! driver_status=$(sed -n 's/.*Driver setup status=//p' $da_log_dir/service.log.1); then
         fail_test "driver setup status=0 not found"
+    else
+        if [ $driver_status -ne 0 ]; then
```

Comment:
Because `fail_test` in the then clause terminates the function, there is no need to protect this if condition in an else clause. FYI, there is an `elif` keyword available to create if-then-elif-then-elif-...-else decision ladders without incurring the readability penalty of deeply nested else clauses.

`[` will error ungracefully if `$driver_status` is an empty string, a word that is not all digits ("abc"), multiple words ("a b"). To prevent these types of failures, use `"$driver_status" == 0` to do a string comparison and ensure the shell treats the value of `$driver_status` as excctly one command line token.